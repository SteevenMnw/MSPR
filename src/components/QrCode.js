import React, { useState, useEffect } from "react";
import { Text, View, StyleSheet, Button, Image } from "react-native";
import { BarCodeScanner } from "expo-barcode-scanner";
import Constants from "expo-constants";
import { Dimensions } from "react-native";

import ModalCoupon from "./ModalCoupon";

const { width } = Dimensions.get("window");
const qrSize = width * 0.7;

export default function QrCode() {
  const [hasPermission, setHasPermission] = useState(null);
  const [scanned, setScanned] = useState(false);
  const [result, setResult] = useState("");
  const [showModal, setShowModal] = useState(false);

  useEffect(() => {
    (async () => {
      const { status } = await BarCodeScanner.requestPermissionsAsync();
      setHasPermission(status === "granted");
    })();
    if (result != "") {
      setShowModal(true);
    }
  }, [result, showModal]);

  const handleBarCodeScanned = ({ type, data }) => {
    setScanned(true);
    setResult(data);
  };

  if (hasPermission === null) {
    return <Text>Requesting for camera permission</Text>;
  }
  if (hasPermission === false) {
    return <Text>No access to camera</Text>;
  }

  return (
    <View
      style={{
        flex: 1,
        flexDirection: "column",
        justifyContent: "flex-end",
      }}
    >
      <BarCodeScanner
        onBarCodeScanned={scanned ? undefined : handleBarCodeScanned}
        style={[StyleSheet.absoluteFillObject, styles.container]}
      >
        <Text style={styles.description}>Scanner votre QR Code</Text>
        <Image style={styles.qr} source={require("../../assets/img/Qr.png")} />
        {scanned && (
          <Button
            title={"Tap to Scan Again"}
            onPress={() => {
              setScanned(false);
              setResult("");
              setShowModal(false);
            }}
          />
        )}
        <Text
          onPress={() => alert("Navigate back from here")}
          style={styles.cancel}
        >
          Cancel
        </Text>
      </BarCodeScanner>
      {showModal ? <ModalCoupon visible={showModal} idCoupon={result} /> : null}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    paddingTop: Constants.statusBarHeight,
    backgroundColor: "#ecf0f1",
    padding: 8,
  },
  qr: {
    marginTop: "20%",
    marginBottom: "20%",
    width: qrSize,
    height: qrSize,
  },
  description: {
    fontSize: width * 0.09,
    marginTop: "10%",
    textAlign: "center",
    width: "70%",
    color: "white",
  },
  cancel: {
    fontSize: width * 0.05,
    textAlign: "center",
    width: "70%",
    color: "white",
  },
});
