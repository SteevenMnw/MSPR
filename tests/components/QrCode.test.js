import React from "react";
import { render } from "@testing-library/react-native";

import QrCode from "../../src/components/QrCode";

it("renders", () => {
  render(<QrCode />);
});
