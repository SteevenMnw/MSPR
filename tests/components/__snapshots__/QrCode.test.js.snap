// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  testID="test_text_RequestCamera"
>
  Requesting for camera permission
</Text>
`;
